src/wmtk/attribute/AttributeScopeHandle.hpp

```diff
@@ -2,14 +2,40 @@
 
 namespace wmtk::attribute {
 struct AttributeManager;
+
+/**
+ * This handle is a wrapper for the MeshManager scope funtions.
+ * It's main task is to enable RAII for attribute scopes.
+ *
+ * It is NOT a handle for `AttributeScope`.
```

Comment:
in abstract it is - this "handle" effectively corresponds to the lifetime of some AttributeScope - the implementation just doesn't track which scope. But yeah we could rename it / make the impl a bit more aware of which scope it is supposed to track?